src/StaticFAQ/index.js

```diff
@@ -74,8 +90,15 @@ class StaticFAQ extends React.Component<Props, State> {
 
     return (
       <SearchState.Consumer>
-        {({ searchText, changeSearchText }: SearchStateType) => {
+        {({
+          searchText,
+          changeSearchText,
+          incrementQueriesCount,
+          resetQueriesCount,
+        }: SearchStateType) => {
           this.changeSearchText = changeSearchText;
+          this.resetQueriesCount = resetQueriesCount;
```

Comment:
this black magic is not needed if you wrap the whole component in another component that pass this as prop

Reply:
Good point, it does feel rather hacky. However if I pass those context functions as props, I will have to add them to the "raw" component's `Props` type that is receiving them, which would maybe defy the purpose of separating the tracking logic from the markup?

Reply:
I can also just create a neutral `onSearch` prop :slightly_smiling_face:

Reply:
This is the worst thing about `context`, if anybody has a better solution I would love to see it 😁

Reply:
The new refactor should work around it! :smile: But your approach is good imho for necessary cases.

Reply:
you had to create new components because of that, that's true, but now it's much better, I swear I can even smell the flowers & herbs while looking at it.